pkg/actors/actor.go

```diff
@@ -93,10 +107,12 @@ func (a *actor) lock(reentrancyID *string) error {
 func (a *actor) unlock() {
 	pending := a.pendingActorCalls.Dec()
 	if pending == 0 {
+		a.disposeLock.Lock()
 		if !a.disposed && a.disposeCh != nil {
 			a.disposed = true
 			close(a.disposeCh)
 		}
+		a.disposeLock.Unlock()
```

Comment:
Suggest to use defer here for unlock(), since close() method is not safe enough (if close() panic, we can't unlock).

Reply:
Updated. Please review again.